src/precice/impl/SolverInterfaceImpl.hpp

```diff
@@ -45,41 +45,29 @@ class Mesh;
 
 namespace impl {
 
-/// Implementation of solver interface.
+/// Implementation of SolverInterface. See also pimpl ideom (https://en.cppreference.com/w/cpp/language/pimpl).
 class SolverInterfaceImpl {
 public:
+  ///@name Construction and Configuration
+  ///@{
+
   /**
-   * @brief Constructor.
+   * @copybrief SolverInterface::SolverInterface(const std::string&, const std::string&, int, int)
    *
    * A solver that wants to use the SolverInterfaceImpl must instantiate an object
    * of this class. The object has to be configured by one of the configure
    * methods before it has a reasonable state and can be used.
    *
-   * @param[in] configurationFileName Name (with path) of the xml config file.
-   * @param[in] participantName Name of the participant using the interface. Has to
-   *                            match the name given for a participant in the
-   *                            xml configuration file.
+   * @copydetails SolverInterface::SolverInterface(const std::string&, const std::string&, int, int)
```

Comment:
This looks a bit messy, but because the middle paragraph is explicitly mentioning `SolverInterfaceImpl` I did not just want to move it inside `SolverInterface`. Is this comment needed? If we explicitly mention the class, it makes copying the documentation more complicated or impossible. Could this be moved to the brief description?

Reply:
I decided to remove most of this part, because it is very outdated. I just kept the part `Use the parameter communicator to specify a custom global MPI communicator. Pass a null pointer to signal preCICE to use MPI_COMM_WORLD.` However, this only shows up in `SolverInterfaceImpl`, because this is only interesting for developers. Users should not pass a `nullptr`, but use the constructor without the `communicator` argument. Is this correct? @uekerman @fsimonis 

A sidenote: I would actually not allow to call `SolverInterface::SolverInterface(const std::string &participantName, const std::string &configurationFileName, int solverProcessIndex, int solverProcessSize, void *communicator)` with `communicator=nullptr`, because this is not the way the public interface is supposed to be used as far as I understand it. Am I missing some context here? We could also open an issue on this and change it with the next breaking release.

Reply:
Yes, this should only be used by user who actually needs to provide a MPI_Comm

Reply:
I will open an issue on this. We should not resolve it here, because this is not a documentation issue anymore.

Reply:
I opened an issue here: https://github.com/precice/precice/issues/1260